### PR TITLE
[APPS-3152] fix for mnt-24137

### PR DESCRIPTION
**Pull request by SathishK-T**
https://hyland.atlassian.net/browse/APPS-3129

This issue is happening when appNamePair shortvalue is generated with stringlowercase irrespective of caseSensitive flag, but as per the scenario the appName in audit application is differed by case (Upper vs Lower). 

When its try to get appNamePair for appName its not matching value and its returning null from convertFromRestAuditQueryParameters(parameters) in getAuditEntriesCountByAppAndProperties method in AuditDAOImpl class.

So instead of getting auditEntries by the appKey we are now getting auditEntries by appName it will resolve the issue. Same approch already was there in getAuditEntry method in AuditImpl class.

Tested Scenarios:

1. name="TESTPermissionAudit" key="testpermissionaudit"
2. name="testpermissionaudit" key="testpermissionaudit"
3. name="testpermissionaudit" key="test"
4. name="test" key="testpermissionaudit"

We can’t use same AppName or AppKey for multiple applications which is already validated. 
If we use same AppName or AppKey for multiple applications we are getting exception like org.alfresco.repo.audit.model.AuditModelException: 11200000 Audit application key 'testpermissionaudit' is used by: AuditApplication[ name=TESTPermissionaudit, id=3, disabledPathsId=4]

### CODE REVIEW

**Review by purusothaman-mm — APPROVED** [review]
LGTM

**Review by vedantmehra9 — APPROVED** [review]

**Review by code4uuuu — APPROVED** [review]
LGTM!

**Review by kavitshah-gl — APPROVED** [review]
Code seems to be okay considering the test that it pass.

**Review by jottley — APPROVED** [review]

File: remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java [hunk 1]
```diff
@@ -913,7 +913,7 @@ public int getAuditEntriesCountByAppAndProperties(AuditService.AuditApplication
         final String applicationName = auditApplication.getKey().substring(1);
```
Comment (tiagosalvado10) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java:913]: Maybe we can change this line to:

`final String applicationName = auditApplication.getName();`

File: remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java [hunk 2]
```diff
@@ -913,7 +913,7 @@ public int getAuditEntriesCountByAppAndProperties(AuditService.AuditApplication
...
 
         AuditQueryParameters parameters = new AuditQueryParameters();
-        parameters.setApplicationName(applicationName);
+        parameters.setApplicationName(auditApplication.getName());
```
Comment (tiagosalvado10) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java:916]: Following the previous suggestion, we could supply the `applicationName` variable again here.

**Review by tiagosalvado10 — CHANGES_REQUESTED** [review]

**Comment by tiagosalvado10** [comment]
I have added two suggestions in order to avoid having two different logic for the application name inside the same method.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java (hunk 1 above)
Comment (SathishK-T) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java]: Changes are done as per the comment.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java (hunk 2 above)
Comment (SathishK-T) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/AuditImpl.java]: Changes are done as per the comment.

**Pull request merged**